### PR TITLE
Create initial API

### LINKED ISSUE #1: Set up api calls to MongoDB in route.ts
## Description

Set up initial API for basic CRUD operations involving the MongoDB Database. 

## Key Features
- Create new events
- Read existing events
- Add volunteers
- Delete single events

### PR DESCRIPTION
**Pull request by intelagense**
## Description

Creates RESTful API using Next.js and establishes a connection to a MongoDB database

### Key Changes:

- API Creation: Developed API endpoints for creating, reading, updating, and deleting events and their associated volunteer information.
- Database Integration: Integrated MongoDB into the API using Mongoose.
- Testing: Tested each API endpoint using Postman to ensure proper handling of diverse scenarios, including valid and invalid data, as well as error responses.

### API Endpoints:
- Create Event (POST /api/events): Endpoint for creating new events with associated volunteer information. 
- Get Event (GET /api/events/:eventId): Endpoint for retrieving detailed information about a specific event.
- Update Event (PUT /api/events/:eventId): Endpoint for updating event information, including volunteers.
- Delete Event (DELETE /api/events/:eventId): Endpoint for deleting an event and its associated data.
(Note: API route.ts currently uses catch-all routing for simplicity and will be updated in a future PR see issue #4 )

Closes #1

### CODE REVIEW

File: src/app/models/event.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import mongoose from "mongoose";
+import { mongoose, models } from 'mongoose';
 
 const EventSchema = new mongoose.Schema({
     event_id: { type: String, required: true, unique: true },
```
Comment (Caleb-Cohen) [inline src/app/models/event.js:4]: Not part of this PR, but the only thing you're storing in the event object is the id? Also, won't mongoose automatically assign an `_id`?

File: src/app/api/[...events]/route.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            { $push: { volunteers: volunteer } },
+            { new: true }
+        );
+        if (!result) {
+            return { status: 404, text: 'Event not found' };
+        }
+
+        return { status: 204 };
+    } catch (error) {
+        console.error("Error deleting event:", error);
```
Comment (Caleb-Cohen) [inline src/app/api/[...events]/route.ts]: Error deleting event in a PUT request?

File: src/app/events/[eventId]/edit/page.tsx [hunk 3]
```diff
@@ -1,11 +1,10 @@
 'use client'
 
-export default function edit() {
+export default function edit( { params }: { params: { eventId:string } }) {
     const makeApiCall = async () => {
 
-        await fetch('/api', {
+        await fetch(`/api/events/${params.eventId}`, {
             method: 'GET',
```
Comment (Caleb-Cohen) [inline src/app/events/[eventId]/edit/page.tsx]: Function is called edit, but the method is a `GET`?

**Review by Caleb-Cohen — COMMENTED** [review]
Just some initial thoughts.

File: src/app/models/event.js (hunk 1 above)
Comment (intelagense) [reply src/app/models/event.js:4]: I think this is confusing as well. 
I will be generating a unique 6-character hash for each event to serve as the URL endpoint.
My thought was to use this endpoint as a custom id. When the user goes to `/events/123abc` the router will use that hash to look up the event.

File: src/app/api/[...events]/route.ts (hunk 2 above)
Comment (intelagense) [reply src/app/api/[...events]/route.ts]: Whoops, copy and paste error.

File: src/app/events/[eventId]/edit/page.tsx (hunk 3 above)
Comment (intelagense) [reply src/app/events/[eventId]/edit/page.tsx]: This will be changed. I was misusing that endpoint just for testing. I will add a comment.

File: src/app/models/event.js (hunk 1 above)
Comment (intelagense) [reply src/app/models/event.js:4]: Oh, and the full schema is in the file. I think the diff is cutting it off.
